src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,138 @@
+import {
+  ChangeEvent,
+  FC,
+  FormEvent,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import cn from 'classnames';
+
+import { Todo } from '../../types/Todo';
+
+import { DispatchContext } from '../../store/store';
+
+interface Props {
+  todo: Todo;
+}
+
+export const TodoItem: FC<Props> = ({ todo }) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [title, setTitle] = useState(todo.title);
+
+  const dispatch = useContext(DispatchContext);
+
+  const handleCompletedTodo = () => {
+    dispatch({
+      type: 'changeTodo',
+      payload: {
+        ...todo,
+        completed: !todo.completed,
+      },
+    });
+  };
+
+  const handleDeleteTodo = () => {
+    dispatch({ type: 'removeTodo', payload: todo.id });
+  };
+
+  const handleTitleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value.trimStart());
+  };
+
+  const handleTitleSave = () => {
+    const trimmedTitle = title.trim();
+
+    if (!trimmedTitle) {
+      handleDeleteTodo();
+
+      return;
+    }
+
+    dispatch({
+      type: 'changeTodo',
+      payload: {
+        ...todo,
+        title: trimmedTitle,
+      },
+    });
+
+    setIsEditing(false);
+  };
+
+  const handleFormSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+  };
+
+  const handleKeyUp = (event: KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      setIsEditing(false);
+      setTitle(todo.title);
+    } else if (event.key === 'Enter') {
+      handleTitleSave();
+    }
+  };
+
+  useEffect(() => {
+    if (isEditing) {
+      document.addEventListener('keyup', handleKeyUp);
+    }
+    // else {
+    //   document.addEventListener('keyup', handleKeyUp);
+    // }
+
+    return () => {
+      document.removeEventListener('keyup', handleKeyUp);
+    };
+  }, [isEditing, handleKeyUp]);
+
+  return (
+    <div
+      data-cy="Todo"
+      className={cn('todo', {
+        completed: todo.completed,
+      })}
+      onDoubleClick={() => setIsEditing(true)}
+    >
+      <label className="todo__status-label">
```

Comment:
`label` element should be associated with a form control. 
To fix this, you should use the `htmlFor `attribute on the `label` to associate it with the checkbox input.